flake8_class_attributes_order/model_parts_info.py

```diff
@@ -64,28 +64,49 @@ def get_funcdef_type(child_node) -> str:
         'staticmethod': 'static_method',
         'classmethod': 'class_method',
 
+        'protected_property': 'protected_property_method',
+        'protected_cached_property': 'protected_property_method',
+        'protected_staticmethod': 'protected_static_method',
+        'protected_classmethod': 'protected_class_method',
+
         'private_property': 'private_property_method',
         'private_cached_property': 'private_property_method',
         'private_staticmethod': 'private_static_method',
         'private_classmethod': 'private_class_method',
     }
-    for decorator_info in child_node.decorator_list:
+    funcdef = get_funcdef_type_by_decorator_info(child_node, decorator_names_to_types_map)
+    if not funcdef:
+        funcdef = get_funcdef_type_by_node_name(child_node, special_methods_names)
+    return funcdef
+
+
+def get_funcdef_type_by_decorator_info(node, decorator_names_to_types_map: dict[str, str]) -> str | None:
+    for decorator_info in node.decorator_list:
         if (
             isinstance(decorator_info, ast.Name)
             and decorator_info.id in decorator_names_to_types_map
         ):
 
-            if child_node.name.startswith('_'):
-                return decorator_names_to_types_map[f'private_{decorator_info.id}']
+            if node.name.startswith('__'):
+                funcdef = decorator_names_to_types_map[f'private_{decorator_info.id}']
+            elif node.name.startswith('_'):
+                funcdef = decorator_names_to_types_map[f'protected_{decorator_info.id}']
+            else:
+                funcdef = decorator_names_to_types_map[decorator_info.id]
+            return funcdef
+    return None
+
 
-            return decorator_names_to_types_map[decorator_info.id]
+def get_funcdef_type_by_node_name(node, special_methods_names: set[str]) -> str:
     funcdef_type = 'method'
-    if child_node.name in special_methods_names:
-        funcdef_type = child_node.name
-    elif child_node.name.startswith('__') and child_node.name.endswith('__'):
+    if node.name in special_methods_names:
+        funcdef_type = node.name
+    elif node.name.startswith('__') and node.name.endswith('__'):
         funcdef_type = 'magic_method'
-    elif child_node.name.startswith('_'):
+    elif node.name.startswith('__'):
         funcdef_type = 'private_method'
+    elif node.name.startswith('_'):
+        funcdef_type = 'protected_method'
     return funcdef_type
```

Comment:
There is also possible to remove `elif` and `else`.
Instead of `funcdef_type = ...` it's simpler to write `return ...`.

Reply:
Problem with linter error CFQ004

Reply:
Fixed, but added `noqa`.

Suggest to review this rule, because it is too strict for switch-case functions